kore/src/Kore/Internal/Substitution.hs

```diff
@@ -419,7 +433,7 @@ wrap xs = Substitution xs
 -- this unless you are sure you need it.
 unsafeWrap
     :: HasCallStack
-    => Ord variable
+    => InternalVariable variable
     => [(SomeVariable variable, TermLike variable)]
```

Comment:
I see Ana left a "TODO" comment about a posible refactoring on line 431. I could take care of it. What do you think @ana-pantilie @ttuegel ?

Reply:
We already have `unsafeWrapFromAssignments`, and I think a major difference between these two is that `unsafeWrap` doesn't enforce the variable renaming invariant. I believe the question here is if we want to be able to create a `Substitution` like this, even for testing (where `unsafeWrap` is most widely used). @ttuegel what do you think? Is there any reason to keep `unsafeWrap` the way it is?